docs/UserGuide.md

```diff
@@ -42,325 +42,586 @@ Bridal Boss is a **desktop app for managing contacts, optimized for use via a  L
 
-### Adding a person: `add`
+- **Client Parameter (`c/`)**:
+    - In wedding commands, accepts either an index number or a name.
+        - e.g., `c/1` or `c/John Doe` are both valid.
 
-Adds a person to the address book.
+- **Date Format**:
+    - Dates must be specified in `YYYY-MM-DD` format.
+        - e.g., `d/2024-12-31` for December 31st, 2024.
 
-Format: `add n/NAME p/PHONE_NUMBER e/EMAIL a/ADDRESS [r/ROLE] [w/WEDDING_INDEX]...​`
+- **Role Parameter (`r/`)**:
+    - Must be a single-word alphanumeric string (no spaces or special characters).
+        - e.g., `r/photographer` is valid, but `r/wedding planner` is not.
 
-<box type="tip" seamless>
+- **Email Addresses**:
+    - Must follow strict validation rules (see [Validation Rules](#validation-rules)).
 
-**Tips:**
-* A person can have either 0 or 1 role.
-* A person can have any number of wedding jobs (including 0).
-</box>
+- **Phone Numbers**:
+    - Must start with 8 or 9 and be exactly 8 digits long.
 
-Examples:
-* `add n/John Doe p/98765432 e/johnd@example.com a/John street, block 123, #01-01`
-* `add n/Betsy Crowe e/betsycrowe@example.com a/Tanglin Mall #03-11 p/12345678 r/Florist`
-* `add n/Betsy Crowe e/betsycrowe@example.com a/Tanglin Mall #03-11 p/12345678 w/1`
+- **Copying Commands**:
+    - When copying commands that span multiple lines (e.g., from a PDF), ensure that spaces are correctly included.
 
-### Listing all persons : `list`
+---
 
-Shows a list of all persons in the address book.
+### Validation Rules
+
+#### Names
+
+- **Allowed Characters**:
+    - Alphabets, spaces, apostrophes (`'`), and hyphens (`-`).
+- **Restrictions**:
+    - Cannot be blank.
+    - Maximum length of 70 characters.
+- **Examples**:
+    - `John Doe`, `Mary-Jane`, `O'Connor`.
+
+#### Phone Numbers
+
+- **Format**:
+    - Must start with 8 or 9.
+    - Exactly 8 digits long.
+    - Numbers only; no spaces or special characters.
+- **Uniqueness**:
+    - Each phone number must be unique in the system.
+- **Examples**:
+    - `91234567`, `82345678`.
+
+#### Email Addresses
+
+- **Format**:
+    - Must be in the form `local-part@domain`.
+- **Local-part**:
+    - Can contain alphanumeric characters and `+`, `_`, `.`, `-`.
+    - Cannot start or end with a special character.
+- **Domain**:
+    - Must have at least two characters.
+    - Cannot start or end with a hyphen (`-`).
+- **Uniqueness**:
+    - Each email must be unique in the system.
+- **Examples**:
+    - `john@example.com`, `user.name+tag@domain.com`.
+
+#### Roles
+
+- **Format**:
+    - Single-word alphanumeric string.
+- **Restrictions**:
+    - No spaces or special characters.
+    - Case-insensitive for matching.
+- **Examples**:
+    - `photographer`, `florist`, `coordinator`.
+
+#### Wedding Fields
+
+- **Wedding Name**:
+    - Cannot be blank.
+- **Date**:
+    - Must be in `YYYY-MM-DD` format.
+- **Venue**:
+    - Cannot be blank or consist only of whitespace when provided.
```

Comment:
I think can specify that this is true only when "v/" is provided (unless it is already written somewhere, I can't remember), because initially when I read this, I thought venue is compulsory.